release.sh

```diff
@@ -1,74 +1,161 @@
+        -*)
+            usage "Unknown option: $1" "" 1
+            ;;
+        *)
+            case "$NARGS" in
+                0)
+                    VERSION="$1"
+                    NARGS=1
+                    ;;
+                1)
+                    NEXT_VERSION="$1"
+                    NARGS=2
+                    ;;
+                *)
+                    usage "Too many arguments: $1" "" 1
+                    ;;
+            esac
+            ;;
+    esac
+    shift
+done
+
+if [[ $NARGS = 0 ]]; then
+    usage "Missing argument: version number" "" 1
 fi
 
-VERSION=$1
-NEXT_VERSION=$2
+# Some sanity checks up front
+if ! command git diff --stat --exit-code HEAD; then
+    echo "Please make sure you have no uncommitted changes" >&2
+    : $((++INSANE))
+fi
+if ! command npm owner ls katex | grep -q "^$(command npm whoami) <"; then
+    echo "You don't seem do be logged into npm, use \`npm login\`" >&2
+    : $((++INSANE))
+fi
+if [[ $BRANCH != @(v*|master) ]]; then
+    echo "'$BRANCH' does not like a release branch to me" >&2
+    : $((++INSANE))
+fi
 
-if [ -z "$NEXT_VERSION" ]; then
-    PROMPT="About to release $VERSION. Look good? [y/n] "
+if [[ -z "$NEXT_VERSION" ]]; then
+    echo "About to release $VERSION from $BRANCH. "
 else
-    PROMPT="About to release $VERSION and bump master to $NEXT_VERSION-pre. Look good? [y/n] "
+    echo "About to release $VERSION from $BRANCH and bump to $NEXT_VERSION-pre."
 fi
-
-read -r -p "$PROMPT" CONFIRM
-if [ "$CONFIRM" != "y" ]; then
-    exit
+if [[ $INSANE != 0 ]]; then
+    read -r -p "$INSANE sanity check(s) failed, really proceed? [y/n] " CONFIRM
+else
+    read -r -p "Look good? [y/n] " CONFIRM
+fi
+if [[ "$CONFIRM" != "y" ]]; then
+    exit 1
 fi
 
 # Make a new detached HEAD
-git checkout master
+git checkout "$BRANCH"
 git pull
 git checkout --detach
+
+# Build generated files and add them to the repository (for bower)
+git clean -fdx build dist
 make setup dist
-git add dist/
+sed -i.bak -E '/^\/dist\/$/d' .gitignore
+rm -f .gitignore.bak
+git add .gitignore dist/
 
 # Edit package.json and bower.json to the right version (see
 # http://stackoverflow.com/a/22084103 for why we need the .bak file to make
 # this mac & linux compatible)
 sed -i.bak -E 's|"version": "[^"]+",|"version": "'$VERSION'",|' package.json
-sed -i.bak -E 's|"version": "[^"]+",|"version": "'$VERSION'",|' bower.json
-rm -f package.json.bak bower.json.bak
+rm -f package.json.bak
+
+# Update the version number in CDN URLs included in the README files,
+# and regenerate the Subresource Integrity hash for these files.
+node update-sri.js "${VERSION}" README.md contrib/*/README.md
 
 # Make the commit and tag, and push them.
 git add package.json bower.json
 git commit -n -m "v$VERSION"
-git tag "v$VERSION"
+git tag -a "v$VERSION" -m "v$VERSION"
 git push origin "v$VERSION"
 
 # Update npm (bower and cdnjs update automatically)
 npm publish
 
 if [ ! -z "$NEXT_VERSION" ]; then
     # Go back to master to bump
```

Comment:
Next time do the review before the merge… But yes, we should fix this on master, and it doesn't really matter for the 0.7 branch. I guess I'll wait till 0.7.1 is released, just in case there is anything else that needs to be changed.